modules/alignment.nf

```diff
@@ -279,47 +214,85 @@ process mergeAndMarkDuplicates {
         --BARCODE_TAG "RX" \
         --ASSUME_SORT_ORDER coordinate \
         --VALIDATION_STRINGENCY SILENT \
+        --ADD_PG_TAG_TO_READS false \
         -I ${bam} \
         -O ${library}_${barcodes}.md.bam \
         -M ${library}.markdups_log
     """
 }
 
-process bwa_index {
-    /* This pipeline is for internal AND external use.
-     * Attempts to link the reference index. If there is no index
+process genome_index {
+    /* Combined process for genome indexing.
+     * Creates both bwameth aligner index and samtools faidx index.
+     * Attempts to link existing indices. If there is no index
      * we download it from the provided URL.
      * If no index and no URL, User will have to debug.
      */
 
     label 'low_cpu'
-    tag { genome }
-    conda "bioconda::samtools=1.19 bioconda::bwameth=0.2.7"
-    storeDir "bwameth_index"
-
+    tag { genome_basename }
+    conda "bioconda::samtools=1.22 bioconda::bwameth=0.2.7"
+    
     output:
-    path "*.{fa,fai,amb,ann,bwt,pac,sa,c2t}"
+    path "bwameth_index/*.{fa,fai,amb,ann,bwt,pac,sa,c2t}", emit: aligner_files
+    tuple path("genome_index/*.fa"), path("genome_index/*.fai"), emit: genome_index
 
     script:
+    genome_basename = file(params.path_to_genome_fasta).baseName
     """
+    # Create output directories
+    mkdir -p bwameth_index genome_index
+    
     real_genome_file="\$(basename ${params.path_to_genome_fasta})"
-    ln -sf "\$(dirname ${params.path_to_genome_fasta})/\${real_genome_file}"* . 
-
-    if [ ! -f "\${real_genome_file}.bwameth.c2t.bwt" ]; then
+    
+    # Handle genome file (download if URL or link if local)
+    if [ ! -f "bwameth_index/\${real_genome_file}" ]; then
```

Comment:
Why? The code is just creating this empty `bwameth_index` folder. It's not going to be there.